.github/workflows/post_publish.yml

```
@@ -51,3 +51,84 @@ jobs:
     if: ${{ github.event.workflow_run.conclusion == 'failure' }}
     steps:
       - run: echo 'The triggering workflow failed'
+
+  registry-check:
+    runs-on: ubuntu-latest
+    needs: [on-success-or-workflow-dispatch]
+    outputs:
+      latest-version: ${{ steps.registry_latest_ver.outputs.current }}
+    steps:
+      - name: Registry Version Check
+        id: registry_latest_ver
+        shell: bash
+        run: |
+          for i in 1 2
+          do
+            LATEST_VERSION=$(curl -s "https://registry.terraform.io/v2/providers/323/provider-versions/latest" | jq -r '.data.attributes.version')
+            if [[ " ${{ needs.on-success-or-workflow-dispatch.outputs.release-tag }}" != ${LATEST_VERSION} ]]; then
+              sleep 1h
+            else
+              echo "Registry and Github Version matches"
+                echo "current=$LATEST_VERSION >> "$GITHUB_OUTPUT"
+            fi
+          done
+
+  os-version-init:
+    strategy:
```

Adding a name to help eliminate `echo` calls (more info in later comment)

```suggestion
  os-version-init:
    name: Run terraform init On Supported Platforms
    strategy:
```